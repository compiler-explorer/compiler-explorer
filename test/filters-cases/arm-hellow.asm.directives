.Ltext0:
main:
.LFB31:
        @ args = 0, pretend = 0, frame = 0
        @ frame_needed = 0, uses_anonymous_args = 0
.LVL0:
        push    {r3, lr}
.LCFI0:
.LBB6:
.LBB7:
        movs    r0, #1
        movw    r1, #:lower16:.LC0
        movt    r1, #:upper16:.LC0
        bl      __printf_chk
.LVL1:
.LBE7:
.LBE6:
.LBB8:
.LBB9:
        movs    r0, #1
        movw    r1, #:lower16:.LC1
        movt    r1, #:upper16:.LC1
        bl      __printf_chk
.LBE9:
.LBE8:
        movs    r0, #0
        pop     {r3, pc}
.LFE31:
.LC0:
        .ascii  "Hello world\000"
.LC1:
        .ascii  "moo\012\000"
.Letext0:
.Ldebug_info0:
.Ldebug_abbrev0:
.Ldebug_loc0:
.LLST0:
.Ldebug_ranges0:
.Ldebug_line0:
.LASF22:
.LASF9:
.LASF30:
.LASF25:
.LASF5:
.LASF8:
.LASF50:
.LASF35:
.LASF19:
.LASF14:
.LASF21:
.LASF26:
.LASF16:
.LASF6:
.LASF34:
.LASF11:
.LASF53:
.LASF31:
.LASF47:
.LASF46:
.LASF43:
.LASF1:
.LASF4:
.LASF7:
.LASF0:
.LASF44:
.LASF33:
.LASF18:
.LASF42:
.LASF15:
.LASF2:
.LASF13:
.LASF55:
.LASF45:
.LASF36:
.LASF37:
.LASF38:
.LASF39:
.LASF40:
.LASF54:
.LASF3:
.LASF20:
.LASF12:
.LASF10:
.LASF27:
.LASF24:
.LASF48:
.LASF29:
.LASF41:
.LASF17:
.LASF32:
.LASF51:
.LASF23:
.LASF28:
.LASF49:
.LASF52:
