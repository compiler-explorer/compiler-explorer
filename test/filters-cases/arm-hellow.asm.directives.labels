main:
        @ args = 0, pretend = 0, frame = 0
        @ frame_needed = 0, uses_anonymous_args = 0
        push    {r3, lr}
        movs    r0, #1
        movw    r1, #:lower16:.LC0
        movt    r1, #:upper16:.LC0
        bl      __printf_chk
        movs    r0, #1
        movw    r1, #:lower16:.LC1
        movt    r1, #:upper16:.LC1
        bl      __printf_chk
        movs    r0, #0
        pop     {r3, pc}
.LC0:
        .ascii  "Hello world\000"
.LC1:
        .ascii  "moo\012\000"
