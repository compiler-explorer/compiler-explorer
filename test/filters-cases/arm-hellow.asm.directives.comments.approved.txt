{
  "asm": [
    {
      "labels": [],
      "source": null,
      "text": ".Ltext0:"
    },
    {
      "labels": [],
      "source": null,
      "text": "main:"
    },
    {
      "labels": [],
      "source": null,
      "text": ".LFB31:"
    },
    {
      "labels": [],
      "source": null,
      "text": ".LVL0:"
    },
    {
      "labels": [],
      "source": {
        "file": null,
        "line": 2
      },
      "text": "        push    {r3, lr}"
    },
    {
      "labels": [],
      "source": null,
      "text": ".LCFI0:"
    },
    {
      "labels": [],
      "source": null,
      "text": ".LBB6:"
    },
    {
      "labels": [],
      "source": null,
      "text": ".LBB7:"
    },
    {
      "labels": [],
      "source": {
        "file": "/usr/arm-linux-gnueabi/include/bits/stdio2.h",
        "line": 105
      },
      "text": "        movs    r0, #1"
    },
    {
      "labels": [],
      "source": {
        "file": "/usr/arm-linux-gnueabi/include/bits/stdio2.h",
        "line": 105
      },
      "text": "        movw    r1, #:lower16:.LC0"
    },
    {
      "labels": [],
      "source": {
        "file": "/usr/arm-linux-gnueabi/include/bits/stdio2.h",
        "line": 105
      },
      "text": "        movt    r1, #:upper16:.LC0"
    },
    {
      "labels": [],
      "source": {
        "file": "/usr/arm-linux-gnueabi/include/bits/stdio2.h",
        "line": 105
      },
      "text": "        bl      __printf_chk"
    },
    {
      "labels": [],
      "source": null,
      "text": ".LVL1:"
    },
    {
      "labels": [],
      "source": null,
      "text": ".LBE7:"
    },
    {
      "labels": [],
      "source": null,
      "text": ".LBE6:"
    },
    {
      "labels": [],
      "source": null,
      "text": ".LBB8:"
    },
    {
      "labels": [],
      "source": null,
      "text": ".LBB9:"
    },
    {
      "labels": [],
      "source": {
        "file": "/usr/arm-linux-gnueabi/include/bits/stdio2.h",
        "line": 105
      },
      "text": "        movs    r0, #1"
    },
    {
      "labels": [],
      "source": {
        "file": "/usr/arm-linux-gnueabi/include/bits/stdio2.h",
        "line": 105
      },
      "text": "        movw    r1, #:lower16:.LC1"
    },
    {
      "labels": [],
      "source": {
        "file": "/usr/arm-linux-gnueabi/include/bits/stdio2.h",
        "line": 105
      },
      "text": "        movt    r1, #:upper16:.LC1"
    },
    {
      "labels": [],
      "source": {
        "file": "/usr/arm-linux-gnueabi/include/bits/stdio2.h",
        "line": 105
      },
      "text": "        bl      __printf_chk"
    },
    {
      "labels": [],
      "source": null,
      "text": ".LBE9:"
    },
    {
      "labels": [],
      "source": null,
      "text": ".LBE8:"
    },
    {
      "labels": [],
      "source": {
        "file": null,
        "line": 5
      },
      "text": "        movs    r0, #0"
    },
    {
      "labels": [],
      "source": {
        "file": null,
        "line": 5
      },
      "text": "        pop     {r3, pc}"
    },
    {
      "labels": [],
      "source": null,
      "text": ".LFE31:"
    },
    {
      "labels": [],
      "source": null,
      "text": ".LC0:"
    },
    {
      "labels": [],
      "source": null,
      "text": "        .ascii  \"Hello world\\000\""
    },
    {
      "labels": [],
      "source": null,
      "text": ".LC1:"
    },
    {
      "labels": [],
      "source": null,
      "text": "        .ascii  \"moo\\012\\000\""
    },
    {
      "labels": [],
      "source": null,
      "text": ".Letext0:"
    },
    {
      "labels": [],
      "source": null,
      "text": ".Ldebug_info0:"
    },
    {
      "labels": [],
      "source": null,
      "text": ".Ldebug_abbrev0:"
    },
    {
      "labels": [],
      "source": null,
      "text": ".Ldebug_loc0:"
    },
    {
      "labels": [],
      "source": null,
      "text": ".LLST0:"
    },
    {
      "labels": [],
      "source": null,
      "text": ".Ldebug_ranges0:"
    },
    {
      "labels": [],
      "source": null,
      "text": ".Ldebug_line0:"
    },
    {
      "labels": [],
      "source": null,
      "text": ".LASF22:"
    },
    {
      "labels": [],
      "source": null,
      "text": ".LASF9:"
    },
    {
      "labels": [],
      "source": null,
      "text": ".LASF30:"
    },
    {
      "labels": [],
      "source": null,
      "text": ".LASF25:"
    },
    {
      "labels": [],
      "source": null,
      "text": ".LASF5:"
    },
    {
      "labels": [],
      "source": null,
      "text": ".LASF8:"
    },
    {
      "labels": [],
      "source": null,
      "text": ".LASF50:"
    },
    {
      "labels": [],
      "source": null,
      "text": ".LASF35:"
    },
    {
      "labels": [],
      "source": null,
      "text": ".LASF19:"
    },
    {
      "labels": [],
      "source": null,
      "text": ".LASF14:"
    },
    {
      "labels": [],
      "source": null,
      "text": ".LASF21:"
    },
    {
      "labels": [],
      "source": null,
      "text": ".LASF26:"
    },
    {
      "labels": [],
      "source": null,
      "text": ".LASF16:"
    },
    {
      "labels": [],
      "source": null,
      "text": ".LASF6:"
    },
    {
      "labels": [],
      "source": null,
      "text": ".LASF34:"
    },
    {
      "labels": [],
      "source": null,
      "text": ".LASF11:"
    },
    {
      "labels": [],
      "source": null,
      "text": ".LASF53:"
    },
    {
      "labels": [],
      "source": null,
      "text": ".LASF31:"
    },
    {
      "labels": [],
      "source": null,
      "text": ".LASF47:"
    },
    {
      "labels": [],
      "source": null,
      "text": ".LASF46:"
    },
    {
      "labels": [],
      "source": null,
      "text": ".LASF43:"
    },
    {
      "labels": [],
      "source": null,
      "text": ".LASF1:"
    },
    {
      "labels": [],
      "source": null,
      "text": ".LASF4:"
    },
    {
      "labels": [],
      "source": null,
      "text": ".LASF7:"
    },
    {
      "labels": [],
      "source": null,
      "text": ".LASF0:"
    },
    {
      "labels": [],
      "source": null,
      "text": ".LASF44:"
    },
    {
      "labels": [],
      "source": null,
      "text": ".LASF33:"
    },
    {
      "labels": [],
      "source": null,
      "text": ".LASF18:"
    },
    {
      "labels": [],
      "source": null,
      "text": ".LASF42:"
    },
    {
      "labels": [],
      "source": null,
      "text": ".LASF15:"
    },
    {
      "labels": [],
      "source": null,
      "text": ".LASF2:"
    },
    {
      "labels": [],
      "source": null,
      "text": ".LASF13:"
    },
    {
      "labels": [],
      "source": null,
      "text": ".LASF55:"
    },
    {
      "labels": [],
      "source": null,
      "text": ".LASF45:"
    },
    {
      "labels": [],
      "source": null,
      "text": ".LASF36:"
    },
    {
      "labels": [],
      "source": null,
      "text": ".LASF37:"
    },
    {
      "labels": [],
      "source": null,
      "text": ".LASF38:"
    },
    {
      "labels": [],
      "source": null,
      "text": ".LASF39:"
    },
    {
      "labels": [],
      "source": null,
      "text": ".LASF40:"
    },
    {
      "labels": [],
      "source": null,
      "text": ".LASF54:"
    },
    {
      "labels": [],
      "source": null,
      "text": ".LASF3:"
    },
    {
      "labels": [],
      "source": null,
      "text": ".LASF20:"
    },
    {
      "labels": [],
      "source": null,
      "text": ".LASF12:"
    },
    {
      "labels": [],
      "source": null,
      "text": ".LASF10:"
    },
    {
      "labels": [],
      "source": null,
      "text": ".LASF27:"
    },
    {
      "labels": [],
      "source": null,
      "text": ".LASF24:"
    },
    {
      "labels": [],
      "source": null,
      "text": ".LASF48:"
    },
    {
      "labels": [],
      "source": null,
      "text": ".LASF29:"
    },
    {
      "labels": [],
      "source": null,
      "text": ".LASF41:"
    },
    {
      "labels": [],
      "source": null,
      "text": ".LASF17:"
    },
    {
      "labels": [],
      "source": null,
      "text": ".LASF32:"
    },
    {
      "labels": [],
      "source": null,
      "text": ".LASF51:"
    },
    {
      "labels": [],
      "source": null,
      "text": ".LASF23:"
    },
    {
      "labels": [],
      "source": null,
      "text": ".LASF28:"
    },
    {
      "labels": [],
      "source": null,
      "text": ".LASF49:"
    },
    {
      "labels": [],
      "source": null,
      "text": ".LASF52:"
    }
  ],
  "labelDefinitions": {
    ".LC0": 27,
    ".LC1": 29,
    "main": 2
  }
}
